CONTRIBUTING.md

```diff
@@ -0,0 +1,131 @@
+Puppet employees we welcome outside contributions. Be aware that your
+suggested change may not fit in with our future plans, so even very good
+code may not be accepted.
+
+Priority will be given to contributions from Puppet Professional Services
+engineers and Service Delivery Partners actively teaching classes.
+
+There are a few guidelines that we need contributors to follow so that 
+we can have a chance of keeping on top of things.
+
+## Getting Started
+
+* Make sure you have a [Jira account](https://tickets.puppetlabs.com)
+* Make sure you have a [GitHub account](https://github.com/signup/free)
+* Submit a ticket for your issue, assuming one does not already exist.
+  * Clearly describe the issue including steps to reproduce when it is a bug.
+  * Make sure you fill in the earliest version that you know has the issue.
+* Fork the repository on GitHub
+
+## Making Changes
+
+Releases are made from the `release` branch, which should always be in a
+releasable state. `master` is the active development branch and changes 
+are periodically moved to the release branch for acceptance testing.
+
+Before you begin working on an issue or feature, check that it hasn't
+already been addressed in the `master` branch.
+
+* Create a topic branch from where you want to base your work.
+  * This is usually the master branch.
+  * Only target release branches if you are certain your fix must be on that
+    branch.
+  * To quickly create a topic branch based on master; `git checkout -b
+    fix/master/my_contribution master`. Please avoid working directly on the
+    `master` branch.
+* Make commits of logical units.
+* Check for unnecessary whitespace with `git diff --check` before committing.
+* Make sure your commit messages are in the proper format.
+
+````
+    (PUP-1234) Make the example in CONTRIBUTING imperative and concrete
+
+    Without this patch applied the example commit message in the CONTRIBUTING
+    document is not a concrete example.  This is a problem because the
+    contributor is left to imagine what the commit message should look like
+    based on a description rather than an example.  This patch fixes the
+    problem by making the example concrete and imperative.
+
+    The first line is a real life imperative statement with a ticket number
+    from our issue tracker.  The body describes the behavior without the patch,
+    why this is a problem, and how the patch fixes the problem when applied.
+````
+
+* Make sure you have added the necessary tests for your changes.
+* Run _all_ the tests to assure nothing else was accidentally broken.
+
+## Making Trivial Changes
+
+### Documentation
+
+For changes of a trivial nature to comments and documentation, it is not
+always necessary to create a new ticket in Jira. In this case, it is
+appropriate to start the first line of a commit with '(doc)' instead of
+a ticket number.
+
+````
+    (doc) Add documentation commit example to CONTRIBUTING
```

Comment:
lol, I think about half of your comments are from things I copied from the puppet repo. I didn't review it too closely since I figured it would be mostly used outside our team.